plugin-server/src/main/utils.ts

```diff
@@ -34,3 +40,61 @@ export async function runInstrumentedFunction({
         clearTimeout(timeout)
     }
 }
+
+export async function kafkaHealthcheck(
+    kafka: Kafka,
+    statsd?: StatsD,
+    timeoutMs = 20000
+): Promise<[boolean, Error | null]> {
+    let consumer: Consumer | null = null
+    let producer: Producer | null = null
+
+    try {
+        producer = kafka.producer()
+        await producer.connect()
+        await producer.send({
+            topic: 'healthcheck',
+            messages: [
+                {
+                    partition: 0,
+                    value: Buffer.from('healthcheck'),
+                },
+            ],
+        })
+
+        let kafkaConsumerWorking = false
+        consumer = kafka.consumer({
+            groupId: 'healthcheck-group',
+        })
+
+        await consumer.subscribe({ topic: 'healthcheck', fromBeginning: true })
+
+        let timer: Date | null = new Date()
+        await consumer.run({
+            eachMessage: async () => {
+                const oldTimer = timer
+                timer = null
```

Comment:
what's the logic with the timer here?

Reply:
The idea is to make sure we don't track the second message consumed. As I'm interested in the time from consumer setup to first message. The little shuffle here I just realized is stupid because eachMessage will process one by one by default.

The idea was that instead of doing:

1. statsd.timing
2. timer = null

we'd make sure to set the timer to minimize the chance that 2 messages get to the statsd.timing point. 

but yeah as I said this is actually irrelevant and confusing, will update